core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -327,8 +331,14 @@ protected void validateNoNewDeletesForDataFiles(TableMetadata base, Long startin
 
     for (DataFile dataFile : dataFiles) {
       // if any delete is found that applies to files written in or before the starting snapshot, fail
-      if (deletes.forDataFile(startingSequenceNumber, dataFile).length > 0) {
-        throw new ValidationException("Cannot commit, found new delete for replaced data file: %s", dataFile);
+      DeleteFile[] deleteFiles = deletes.forDataFile(startingSequenceNumber, dataFile);
+      if (ignoreEqualityDeletes) {
+        ValidationException.check(Arrays.stream(deleteFiles)
+                .noneMatch(deleteFile -> deleteFile.content() == FileContent.POSITION_DELETES),
```

Comment:
Can you wrap the line after `check`? I think it looks strange to have a double indent because the first part of the line wasn't wrapped.

Reply:
done